dot/build_spec_test.go

```diff
@@ -0,0 +1,286 @@
+// Copyright 2021 ChainSafe Systems (ON)
+// SPDX-License-Identifier: LGPL-3.0-only
+
+package dot
+
+import (
+	"errors"
+	"os"
+	"path/filepath"
+	"strings"
+	"testing"
+
+	"github.com/ChainSafe/gossamer/lib/genesis"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+)
+
+func TestBuildSpec_ToJSON(t *testing.T) {
+	tests := []struct {
+		name    string
+		genesis *genesis.Genesis
+		want    string
+		err     error
+	}{
+		{
+			name:    "name test",
+			genesis: &genesis.Genesis{Name: "test"},
+			want: `
+{
+    "name": "test",
+    "id": "",
+    "chainType": "",
+    "bootNodes": null,
+    "telemetryEndpoints": null,
+    "protocolId": "",
+    "genesis": {},
+    "properties": null,
+    "forkBlocks": null,
+    "badBlocks": null,
+    "consensusEngine": "",
+    "codeSubstitutes": null
+}`,
+		},
+		{
+			name: "additional parameters test",
+			genesis: &genesis.Genesis{
+				Name:            "test",
+				ID:              "ID",
+				ChainType:       "chainType",
+				ProtocolID:      "protocol",
+				ConsensusEngine: "babe",
+			},
+			want: `
+{
+    "name": "test",
+    "id": "ID",
+    "chainType": "chainType",
+    "bootNodes": null,
+    "telemetryEndpoints": null,
+    "protocolId": "protocol",
+    "genesis": {},
+    "properties": null,
+    "forkBlocks": null,
+    "badBlocks": null,
+    "consensusEngine": "",
+    "codeSubstitutes": null
+}`,
+		},
+		{
+			name: "normal conditions",
+			genesis: &genesis.Genesis{
+				Name:               "test",
+				ID:                 "ID",
+				ChainType:          "chainType",
+				Bootnodes:          []string{"node1", "node2"},
+				TelemetryEndpoints: []interface{}{"endpoint"},
+				ProtocolID:         "protocol",
+				Genesis:            genesis.Fields{},
+				Properties:         map[string]interface{}{"key": "value"},
+				ForkBlocks:         []string{"1", "2"},
+				BadBlocks:          []string{"3", "4"},
+				ConsensusEngine:    "babe",
+				CodeSubstitutes:    map[string]string{"key": "value"},
+			},
+			want: `
+{
+    "name": "test",
+    "id": "ID",
+    "chainType": "chainType",
+    "bootNodes": [
+        "node1",
+        "node2"
+    ],
+    "telemetryEndpoints": null,
+    "protocolId": "protocol",
+    "genesis": {},
+    "properties": {
+        "key": "value"
+    },
+    "forkBlocks": null,
+    "badBlocks": null,
+    "consensusEngine": "",
+    "codeSubstitutes": null
+}`,
+		},
+	}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			b := &BuildSpec{
+				genesis: tt.genesis,
+			}
+			got, err := b.ToJSON()
+			assert.ErrorIs(t, err, tt.err)
+			assert.Equal(t, strings.TrimSpace(tt.want), string(got))
+		})
+	}
+}
+
+func TestBuildFromDB(t *testing.T) {
+	// initialise node (initialise state database and load genesis data)
+	cfg := NewTestConfig(t)
+	cfg.Init.Genesis = "../chain/gssmr/genesis.json"
+	nodeInstance := nodeBuilder{}
+	err := nodeInstance.initNode(cfg)
+	assert.NoError(t, err)
```

Comment:
We want to stop the test if this fails

```suggestion
	require.NoError(t, err)
```

Reply:
Updated